src/transformers/models/atlas/configuration_atlas.py

```
@@ -0,0 +1,177 @@
+            Whether or not to disregard the BOS token when computing the loss.
+        output_retrieved(`bool`, *optional*, defaults to `False`):
+            If set to `True`, `retrieved_doc_embeds`, `retrieved_doc_ids`, `context_input_ids` and
+            `context_attention_mask` are returned. See returned tensors for more detail.
+        use_cache (`bool`, *optional*, defaults to `True`):
+            Whether or not the model should return the last key/values attentions (not used by all models).
+        forced_eos_token_id (`int`, *optional*):
+            The id of the token to force as the last generated token when `max_length` is reached. Usually set to
+            `eos_token_id`.
+"""
+
+
+@add_start_docstrings(ATLAS_CONFIG_DOC)
+class AtlasConfig(PretrainedConfig):
+    model_type = "atlas"
+    is_composition = True
+
+    def __init__(
+        self,
+        vocab_size=None,
+        is_encoder_decoder=True,
+        prefix=None,
+        bos_token_id=None,
+        pad_token_id=None,
+        eos_token_id=None,
+        decoder_start_token_id=None,
+        n_docs=5,
+        max_combined_length=512,
+        do_deduplication=True,
+        output_retrieved=False,
+        use_cache=True,
+        closed_book=False,
+        forced_eos_token_id=None,
+        query_side_retriever_training=False,
+        temperature_gold = 0.01,
+        temperature_score = 0.01,
+        **kwargs
+    ):
+        super().__init__(
+            bos_token_id=bos_token_id,
+            pad_token_id=pad_token_id,
+            eos_token_id=eos_token_id,
+            decoder_start_token_id=decoder_start_token_id,
+            forced_eos_token_id=forced_eos_token_id,
+            is_encoder_decoder=is_encoder_decoder,
+            prefix=prefix,
+            vocab_size=vocab_size,
+            **kwargs,
+        )
+        assert (
+            "retriever" in kwargs and "generator" in kwargs
+        ), "Config has to be initialized with retriever and generator config"
```

No assert, let's raise an appropriate error